astronomer/providers/amazon/aws/operators/batch.py

```diff
@@ -7,14 +7,17 @@
     - `Batch <http://boto3.readthedocs.io/en/latest/reference/services/batch.html>`_
     - `Welcome <https://docs.aws.amazon.com/batch/latest/APIReference/Welcome.html>`_
 """
-from typing import Any, Dict
+from typing import TYPE_CHECKING, Any, Dict
 
 from airflow.exceptions import AirflowException
 from airflow.providers.amazon.aws.operators.batch import BatchOperator
-from airflow.utils.context import Context
 
 from astronomer.providers.amazon.aws.triggers.batch import BatchOperatorTrigger
 
+if TYPE_CHECKING:
+    # TODO: This can be removed from TYPE_CHECKING once there is a minimum requirement of Airflow 2.2.3+
+    from airflow.utils.context import Context
```

Comment:
We have one more option:

1. Have a `compat` module in this repo and import `airflow.utils.context.Context` if it is available, or just set `Context = Dict` if it is not available, so it is consistent.


wydt?

Reply:
I like the idea of having dynamic check using `compat` module.

@josh-fell could you please make the change and fix the code coverage.

Reply:
> Have a compat module in this repo and import airflow.utils.context.Context if it is available, or just set Context = Dict if it is not available, so it is consistent.

+1 ❤️  this idea as well. I'll add a `compat` module and unify the typing `Context` across the repo.